.gitignore

```diff
@@ -112,3 +112,4 @@ venv.bak/
 *.pkl
 
 .DS_Store
+poetry.lock
```

Comment:
I've generally seen this committed, any reason why you want to ignore it?

Reply:
So, Poetry says that committing the `poetry.lock` file is optional if your python package is a library (as opposed to a webserver package you're deploying or whatever) https://python-poetry.org/docs/libraries/#lock-file

My reasoning was that if I don't include the lock file, we get updated dependencies on every test run and can see if a dependency upgrade has broken things, letting us know that we should lower the upper bound of our dependencies.

But happy to commit it so developers on the project have the same requirements!

Reply:
That makes sense, lets leave it out for now